### PR TITLE
feat(tests) add e2e test

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by lucasfernog** [comment]
When did we stop shipping updater.rs? 
But we really need to remove it on the alpha release.

**Comment by tensor-programming** [comment]
> When did we stop shipping updater.rs?
> But we really need to remove it on the alpha release.

We should have stopped when we moved updater into its own crate.  Since we aren't including the updater logic by default should also make it optional or just remove it until we make the updater logic work.

**Comment by nothingismagick** [comment]
How is this coming along?

**Comment by lucasfernog** [comment]
I need some help with Jest. I can't get this async test to work.

**Comment by nklayman** [comment]
I'll give it a shot tonight or tomorrow. Which one is it?

**Comment by nklayman** [comment]
I made some progress, but will have to finish tomorrow.

**Comment by lucasfernog** [comment]
@nklayman sorry I had to break the TS API again :D I can update the vue-cli-plugin after we publish tauri.js again.

File: cli/tauri.js/test/jest/__tests__/build.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      let appPid
+      const server = fixtureSetup.startServer(() => appPid, resolve)
+      const result = build(tauriConfig)
+      await result.promise
+
+      const artifactFolder = tauriConfig.ctx.debug ? 'debug' : 'release'
+      const artifactPath = path.resolve(appDir, `src-tauri/target/${artifactFolder}/app`)
+
+      appPid = spawn(
+        process.platform === 'win32' ? `${artifactPath}.exe` : artifactPath.replace(`${artifactFolder}/app`, `${artifactFolder}/./app`),
```
Comment (nklayman) [inline cli/tauri.js/test/jest/__tests__/build.spec.js]: Why is this necessary on windows? Maybe add a comment so it doesn't get removed on accident.

File: cli/tauri.js/test/jest/fixtures/app-test-setup.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const appDir = path.resolve(__dirname, '../fixtures/app')
+const tauriDir = path.join(appDir, 'src-tauri')
+
+module.exports.appDir = appDir
+module.exports.distDir = path.join(appDir, 'dist')
+
+import * as appPaths from '../../../src/helpers/app-paths'
+
+module.exports.initJest = () => {
+  jest.mock('helpers/non-webpack-require', () => {
```
Comment (nklayman) [inline cli/tauri.js/test/jest/fixtures/app-test-setup.js]: You should add a comment in helpers/non-webpack-require explaining that it is a helper so that it can be mocked.

File: cli/tauri.js/test/jest/__tests__/build.spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  })
+}
+
+describe('Tauri Build', () => {
+  const build = {
+    devPath: distDir,
+    distDir: distDir
+  }
+
+  for (const debug of [true, false]) {
```
Comment (nklayman) [inline cli/tauri.js/test/jest/__tests__/build.spec.js]: It may be better to use [test.each](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout).

File: cli/tauri.js/test/jest/__tests__/dev.spec.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+describe('Tauri Dev', () => {
+  const build = {
+    distDir: distDir
+  }
+
+  const devServer = startDevServer()
+
+  for (const devPath of [devServer.url, distDir]) {
```
Comment (nklayman) [inline cli/tauri.js/test/jest/__tests__/dev.spec.js]: It may be better to use [test.each](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout).

File: cli/tauri.js/package.json [hunk 5]
```diff
@@ -87,9 +92,11 @@
     "eslint-plugin-promise": "4.2.1",
     "eslint-plugin-security": "1.4.0",
     "eslint-plugin-standard": "4.0.1",
-    "husky": "4.0.10",
+    "express": "^4.17.1",
+    "husky": "3.1.0",
```
Comment (nklayman) [inline cli/tauri.js/package.json]: Why was husky dropped in version?

File: cli/tauri.js/test/jest/fixtures/app-test-setup.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  appPaths.appDir = appDir
+  appPaths.tauriDir = tauriDir
+  jest.spyOn(appPaths.resolve, 'app').mockImplementation(dir => path.resolve(appDir, dir))
+  jest.spyOn(appPaths.resolve, 'tauri').mockImplementation(dir => path.resolve(tauriDir, dir))
+}
+
+module.exports.startServer = (getAppPid, onReply) => {
+  const express = require('express')
+  const cors = require('cors')
+  const app = express()
```
Comment (nklayman) [inline cli/tauri.js/test/jest/fixtures/app-test-setup.js]: Express seems really overkill for this, why not use the [http.createServer](https://nodejs.org/api/http.html#http_http_createserver_options_requestlistener) included with node?

File: cli/tauri.js/test/jest/__tests__/build.spec.js [hunk 7]
```diff
@@ -43,35 +43,37 @@ describe('Tauri Build', () => {
     distDir: distDir
   }
 
-  for (const debug of [true, false]) {
-    it(`works with the embedded-server ${debug ? 'debug' : 'release'} mode`, () => {
+  test.each([['debug'], ['release']])(
```
Comment (nklayman) [inline cli/tauri.js/test/jest/__tests__/build.spec.js]: You can use `it.each` instead, that would stay in line with the rest of the tests.

File: cli/tauri.js/test/jest/__tests__/build.spec.js [hunk 8]
```diff
@@ -43,35 +43,37 @@ describe('Tauri Build', () => {
...
         }
       })
-    })
-  }
+    }
+  )
 
-  for (const debug of [true, false]) {
-    it(`works with the no-server ${debug ? 'debug' : 'release'} mode`, () => {
+  test.each([['debug'], ['release']])(
```
Comment (nklayman) [inline cli/tauri.js/test/jest/__tests__/build.spec.js]: You can use `it.each` instead, that would stay in line with the rest of the tests.

File: cli/tauri.js/test/jest/__tests__/dev.spec.js [hunk 9]
```diff
@@ -48,9 +57,10 @@ describe('Tauri Dev', () => {
 
   const devServer = startDevServer()
 
-  for (const devPath of [devServer.url, distDir]) {
-    const runningDevServer = devPath.startsWith('http')
-    it(`works with the ${runningDevServer ? 'dev server' : 'dev path'} mode`, () => {
+  test.each([[devServer.url], [distDir]])(
```
Comment (nklayman) [inline cli/tauri.js/test/jest/__tests__/dev.spec.js]: You can use `it.each` instead, that would stay in line with the rest of the tests.

**Comment by lucasfernog** [comment]
Need this merged to fix #493 , so I just made Jest ignore the e2e tests for now.

**Comment by lucasfernog** [comment]
I think the only problem here is that I made some breaking changes to the dev/build API, so vue-cli-plugin will need changes @nklayman

**Comment by nothingismagick** [comment]
that means a minor semver bump across the board in my opinion. maybe a good time to merge in #472

**Comment by jbolda** [comment]
Do you think we could merge dev into this again? it's been a month and we have done quite a bit since

**Comment by lucasfernog** [comment]
Done it already.

**Comment by nothingismagick** [comment]
LGTM 🚀

**Comment by jbolda** [comment]
@lucasfernog read that wrong, my bad 🤦🏻‍♂️
